server/db/models/product.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+var mongoose = require('mongoose');
+var Schema = mongoose.Schema;
+
+var schema =  new mongoose.schema({
+	name: {type: String, required: true},
+	sku: {type: String, required: true, unique: true}
+	desc: String,
+	price: {type: Number, get: getPrice, set: setPrice, required: true},
+	type: {type: String, required: true},
+	store: {type: Schema.Types.ObjectId, ref: 'Store', required: true},
+	images: [{data: Buffer, contentType: String}],
+	composite: [{data: Buffer, contentType: String}],
+	stockQty: {type: Number, required: true, default: 0},
+	colors: {{black: boolean, default: false}, {darkBrown: boolean, default: false}, 
```

Comment:
can we achieve this with enum? having a color as a string and limiting the possibilities